### PR TITLE
#6701 Report storage tests

### PR DESCRIPTION
**Pull request by mg80**
Functional and unit tests

### CODE REVIEW

**Comment by mg80** [comment]
@sulkaharo I added some tests, but it looks like the build is broken due to an unrelated issue

**Comment by sulkaharo** [comment]
Looks like the tests breakage is caused by some change in the tests in this PR interfering with the other tests - the DEV tests run on without these changes but this PR fails both in the cloud and locally. I suspect the issue is with the report storage initialising the local storage which is also used but other tests. I'm not sure I have the time immediately to check what's going on, so if you have time to check what's causing the test to fail and help implement a clean teardown between tests, that'd be great.

**Comment by mg80** [comment]
@sulkaharo wow that was a nasty one. js-storage evaluates if it's running on the window or not as soon as it's required. This becomes a problem when js-storage is required anywhere in the AST before the window object is setup. I'm guessing this wasn't a problem until now as I added a pure unit test.

File: tests/hashauth.test.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ describe('hashauth', function ( ) {
   });
 
   after(function (done) {
+    // cleanup js-storage as it evaluates if the test is running in the window or not when first required
+    delete require.cache[require.resolve('js-storage')];
```
Comment (sulkaharo) [inline tests/hashauth.test.js:20]: 😍👍

**Comment by sulkaharo** [comment]
GREAT catch! Makes me wonder if we should have a generic teardown include file that deletes the require cache for this and other core requires between all tests. 🤔

**Pull request merged**